File: PlacementCellBackend/Services/CRUD/Interfaces/ICollageService.cs
using PlacementCellBackend.Models;

namespace PlacementCellBackend.Services.CRUD.Interfaces;

public interface ICollageService
{
    Task<IEnumerable<College>> GetAllCollegesAsync();
    Task<College?> GetCollegeByIdAsync(int id);
    Task<College> CreateCollegeAsync(College college);
    Task<bool> UpDateCollegeAsync(int id, College college);
    Task<bool> DeleteCollegeAsync(int id);
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/CRUD/Interfaces/ICollageService.java
package com.example.placementicsbackend.services.crud.interfaces;

import com.example.placementicsbackend.models.College;
import java.util.List;

public interface ICollageService {

    List<College> getAllColleges();

    College getCollegeById(int id);

    College createCollege(College college);

    boolean updateCollege(int id, College college);

    boolean deleteCollege(int id);
}
